lib/Sema/TypeCheckMacros.cpp

```diff
@@ -1000,6 +1030,34 @@ bool swift::expandMembers(CustomAttr *attr, MacroDecl *macro, Decl *decl) {
   return synthesizedMembers;
 }
 
+void swift::expandPeers(CustomAttr *attr, MacroDecl *macro, Decl *decl,
+                        SmallVectorImpl<Decl *> &peers) {
+  auto macroSourceFile = evaluateAttachedMacro(macro, decl, attr,
+                                               /*passParentContext*/false,
+                                               MacroRole::Peer);
+  if (!macroSourceFile)
+    return;
+
+  PrettyStackTraceDecl debugStack("applying expanded peer macro", decl);
+
+  auto *parent = decl->getDeclContext();
+  auto topLevelDecls = macroSourceFile->getTopLevelDecls();
+  for (auto peer : topLevelDecls) {
+    peer->setDeclContext(parent);
```

Comment:
I thought the `DeclContext` would already be correct based on us setting the parser's `CurDeclContext` before we parse these top-level declarations?

Reply:
Heh, I think you're right! I wrote this out of habit, not because there was some issue that prompted me to.